modal/functions.py

```diff
@@ -474,11 +473,11 @@ def _set_output_mgr(self, output_mgr: OutputManager):
         """mdmd:hidden"""
         self._output_mgr = output_mgr
 
-    def _set_raw_f(self, raw_f):
-        self._raw_f = raw_f
+    def _set_info(self, function_info: FunctionInfo):
+        self._info = function_info
 
-    def _set_function(self, function: "_Function"):
-        self._function = function
+    def _set_stub(self, stub):
+        self._stub = stub
```

Comment:
would prefer to avoid this – it creates a circular reference between stubs and functions and those things tend to make things harder to refactor

if you have the function's name, you can look it up directly from the stub – that should be enough right?

Reply:
It would be nice to not have it, I agree, but the point of this refactor is to let users specify *only* the function path, and not have to go through the "stub path" in order to access it, e.g. `modal run foofile.py::myfunc`, and then we need a way to look up the stub given only the FunctionHandler 🤔

Reply:
ok, seems acceptable! I can't think of any other simpler ways to accomplish the same thing.